docs/migration-guide.md

```diff
@@ -0,0 +1,330 @@
+8. Rewrite everything using `page.$eval` to access the element directly (just like unit tests do)
+9. Replace calls with `getRect` and `screenshotAndCompare` with `await expect(elem).to.be.golden()`
+10. Make use of the `focusElem`, `clickElem`, `sendKeysElem`, `oneEvent`, etc. helpers where possible
+
+Before (HTML):
+
+```html
+<html>
+  <head>
+    <script type="module">
+      import '@brightspace-ui/core/components/typography/typography.js';
+      import '../my-elem.js';
+    </script>
+  </head>
+  <body class="d2l-typography">
+    <div>
+      <my-elem id="default"></my-elem>
+    </div>
+  </body>
+</html>
+```
+
+Before (JavaScript):
+
+```javascript
+import puppeteer from 'puppeteer';
+import { VisualDiff } from '@brightspace-ui/visual-diff';
+
+describe('my-elem', () => {
+
+  const visualDiff = new VisualDiff('my-elem', import.meta.url);
+
+  let browser, page;
+
+  before(async() => {
+    browser = await puppeteer.launch();
+    page = await visualDiff.createPage(browser);
+    await page.goto(`${visualDiff.getBaseUrl()}/test/my-elem.visual-diff.html`, { waitUntil: ['networkidle0', 'load'] });
+    await page.bringToFront();
+  });
+
+  beforeEach(async() => {
+    await visualDiff.resetFocus(page);
+  });
+
+  after(async() => await browser.close());
+
+  it('passes visual-diff comparison', async function() {
+    await page.$eval('#default', (elem) => {
+      const innerElem = elem.shadowRoot.querySelector('button');
+      button.focus();
+    });
+    const rect = await visualDiff.getRect(page, '#default');
+    await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+  });
+
+});
+```
+
+After:
+
+```javascript
+import '../my-elem.js'
+import { expect, fixture, focusElem, html } from '@brightspace-ui/testing';
+
+describe('my-elem', () => {
+
+  it('passes visual-diff comparison', async() => {
+    const elem = await fixture(html`<my-elem></my-elem>`);
+    await focusElem(elem.shadowRoot.querySelector('button'));
+    await expect(elem).to.be.golden();
+  });
+
+});
+```
+
+Much less code, right?
+
+### Step 3: Test Migration Locally
+
+Test the migration by running `d2l-test-runner vdiff`, filtering by file using `--filter <file-name>` if appropriate.
+
+If tests fail, run `d2l-test-runner report` to view a HTML report visualizing the differences.
+
+Tips:
+
+* If possible, line up the snapshot element size and viewport size to match the old values. This will result in a much clearer picture of any changes - or there may be none at all!
+* The Daylight fonts and typography are now included automatically. If those weren't added previously, this will result in some font changes. It may be helpful to turn these off with inline styles (`font-family: auto; letter-spacing: normal;, maybe color: black; font-size: 20px;`) while developing or to get an initial "clean" CI run to verify there are no actual unexpected changes. They can then be removed before merging the new goldens.
```

Comment:
```suggestion
* The Daylight fonts and typography are now included automatically. If those weren't added previously, this will result in some font changes. It may be helpful to turn these off with inline styles (`font-family: auto; letter-spacing: normal;`, maybe `color: black; font-size: 20px;`) while developing or to get an initial "clean" CI run to verify there are no actual unexpected changes. They can then be removed before merging the new goldens.
```